lib/cn/cn.c

```diff
@@ -810,8 +812,8 @@ cn_kvset_cb(struct cndb_cn_ctx *ctx, struct kvset_meta *km, u64 kvsetid)
         return err;
     }
 
-    if (km->km_dgen > cn_tree_initial_dgen(cn->cn_tree))
-        cn_tree_set_initial_dgen(cn->cn_tree, km->km_dgen);
+    if (km->km_dgen > cn_tree_initial_dgen(tree))
+        cn_tree_set_initial_dgen(tree, km->km_dgen);
```

Comment:
Is it worth having nodes keep track of the max dgen and max key?  That info can be learned in `cn_node_insert_kvset()`.  Then we can more logically move this call to `cn_tree_set_initial_dgen()` into `cn_open()` and also make it easier to build the route map (wouldn't have to walk the node's kvset list).

Reply:
Seems reasonable.  One upside of walking the nodes is to sanity check for duplicates (which I got burned on when there were failures in the commit path).  Open a ticket?

Reply:
Do you wan't to open a ticket for this?

Reply:
A downside of what I proposed is each node would need a max keylen buffer to store the key.  It wouldn't be safe to hold a ptr to the kvset's key after we went live.  Perhaps better to leave it as is.